automation/taskcluster/androidTest/ui-test.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This script does the following:
+# 1. Retrieves glcoud service account token
+# 2. Activates gcloud service account
+# 3. Connects to google Firebase (using TestArmada's Flank tool)
+# 4. Executes UI tests
+# 5. Puts test artifacts into the test_artifacts folder
+
+# NOTE:
+# Flank supports sharding across multiple devices at a time, but gcloud API
+# only supports 1 defined APK per test run.
+
+
+# If a command fails then do not proceed and fail this script too.
+set -e
+
+#########################
+# The command line help #
+#########################
+display_help() {
+    echo "Usage: $0 Build_Variant [Number_Shards...]"
+    echo
+    echo "Examples:"
+    echo "To run UI tests on ARM device shard (1 test / shard)"
+    echo "$ ui-test.sh arm -1"
+    echo
+    echo "To run UI tests on X86 device (on 3 shards)"
+    echo "$ ui-test.sh feature x86 3"
+    echo
+}
+
+# Basic parameter check
+if [[ $# -lt 1 ]]; then
+    echo "Your command line contains $# arguments"
+    display_help
+    exit 1
+fi
+
+device_type="$1"  # arm | x86
+if [[ ! -z "$2" ]]; then
+    num_shards=$2
+fi
+
+JAVA_BIN="/usr/bin/java"
+PATH_TEST="./automation/taskcluster/androidTest"
+FLANK_BIN="/build/test-tools/flank.jar"
+FLANK_CONF_ARM="${PATH_TEST}/flank-arm.yml"
+FLANK_CONF_X86="${PATH_TEST}/flank-x86.yml"
+
+echo
+echo "RETRIEVE SERVICE ACCT TOKEN"
+echo
+python automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/firebase -k firebaseToken  -f $GOOGLE_APPLICATION_CREDENTIALS 
```

Comment:
I believe that we'll need to have separate firebase tokens for each channel (`nightly`/`beta`/`production`)
I'd recommend adding a block to the `project/mobile/fenix/(nightly|nightlyLegacy|beta|production)` secrets

Reply:
Ah, this is an on-pr check.
In that case, since the credential is essentially public, since it's run on PRs ("staging", kinda), maybe throw the credentials in `garbage/staging/project/mobile/fenix`?

Reply:
> Ah, this is an on-pr check.
> In that case, since the credential is essentially public, since it's run on PRs ("staging", kinda), maybe throw the credentials in `garbage/staging/project/mobile/fenix`?

thanks for the review! this is actually an on github push, see: https://github.com/rpappalax/fenix/blob/add-ui-tests-to-taskcluster/automation/taskcluster/decision_task.py#L81-L97  - Also, is there some advantage to making this key public?  I'm not sure I'd feel comfortable doing that unless there was good reason

Reply:
Since the key is used in level 1 builds (any branch push, and it looks like PR too), it's possible for the key to be leaked by a malicious attacker adjusting the automation.

I'm worried that putting the secrets in non-`garbage` implies that the secrets can be trusted to be private (which isn't the case). So, other, _real_ secret values could be added to the same `project/mobile/fenix/firebase` group under the assumption that they're safe there (even though they're not).

Note that there _is_ downside to having these credentials in `garbage` - they're just a click away from usage, rather than requiring contributors to modify automation to leak them. So, we lose some "security by obscurity", but at least it means that more critical secrets won't be accidentally leaked in the future.

-----

Perhaps we could leave the secret in non-`garbage`, but not run ui tests on-PR? Or, alternatively, run UI tests without the extra credentials on-PR, somehow?

Reply:
@mitchhentges i believe @JohanLorenzo may have already addressed this issue in a few other project (firefox-tv, a-c and r-b)?

Reply:
I see where the confusion comes from. @rpappalax, the code of the decision task (in this PR) is different from the one in r-b. The current PR schedules a UI job on PRs, which is why the TC task is currently red https://tools.taskcluster.net/groups/JsXobNS9SdqPJvGmgvsLMw/tasks/JsXobNS9SdqPJvGmgvsLMw/runs/0/logs/public%2Flogs%2Flive.log#L851. 

If my recollection is correct, we cannot put a real token in `garbage` because this token can be used by anybody opening a PR. That's the reason why `project/mobile/fenix/firebase` is only available on the main repo and @rpappalax' fork https://hg.mozilla.org/ci/ci-configuration/file/a133ce097f7e44850e45561263e92287eeb485a6/grants.yml#l1159. We currently allow all branches for the sake of initial development. Eventually, I think we should restrict the scope to master on the main repo + 1 fork (it can be any branches there). 

Regarding the use of `project/mobile/fenix/(nightly|nightlyLegacy|beta|production)`, these TC secrets were added on Fenix https://github.com/mozilla-mobile/fenix/pull/3187, at the same time as the UI tests were added on R-B https://github.com/mozilla-mobile/reference-browser/pull/790. I think we need to uniformize them both. I see some tokens are duplicated in all 4 secrets. We can duplicate the Firebase token the same way, for now. Although, I wonder if we should place all identical secrets in something like `project/mobile/fenix/common`. This can be applied to reference-browser too.  What do you think @mitchhentges?

Reply:
> I see where the confusion comes from. @rpappalax, the code of the decision task (in this PR) is different from the one in r-b. The current PR schedules a UI job on PRs, which is why the TC task is currently red https://tools.taskcluster.net/groups/JsXobNS9SdqPJvGmgvsLMw/tasks/JsXobNS9SdqPJvGmgvsLMw/runs/0/logs/public%2Flogs%2Flive.log#L851.
> 
@JohanLorenzo ah, that was an oversight on my part.  It should only be on push(). I was doing on pr() too in my own branch to get the jobs to run more frequently.

> If my recollection is correct, we cannot put a real token in `garbage` because this token can be used by anybody opening a PR. That's the reason why `project/mobile/fenix/firebase` is only available on the main repo and @rpappalax' fork https://hg.mozilla.org/ci/ci-configuration/file/a133ce097f7e44850e45561263e92287eeb485a6/grants.yml#l1159. We currently allow all branches for the sake of initial development. Eventually, I think we should restrict the scope to master on the main repo + 1 fork (it can be any branches there).
> 
Eventually, everyone on our team that is working on Fenix UI tests will need to be able to run tc jobs on their own fork. We currently have 5 of us, but I'm the only one with scopes to do this atm which is making UI test dev work fairly cumberson.  I'll follow-up w/ you on this in a pm.